src/global_context.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "entity/mesh.hpp"
+#include "terrain/chunk.hpp"
+
+#include <map>
+#include <set>
+#include <mutex>
+
+#include "thread-pool/include/BS_thread_pool.hpp"
```

Comment:
```suggestion
#include <BS_thread_pool.hpp>
```

And then in CMakeLists.txt

```cmake
target_include_directories(
    SYSTEM PRIVATE
    "vendor/thread-pool/include"
)